public/src/client/topic.js

```diff
@@ -197,10 +198,29 @@ define('forum/topic', [
 			}
 		}
 
-		var currentBookmark = localStorage.getItem('topic:' + ajaxify.data.tid + ':bookmark');
+		var bookmarkKey = 'topic:' + ajaxify.data.tid + ':bookmark';
+		var currentBookmark = ajaxify.data.bookmark || localStorage.getItem(bookmarkKey);
 
+		if (!currentBookmark || parseInt(postIndex, 10) > parseInt(currentBookmark, 10)) {
+			if (app.user.uid) {
+				var data = {
+					'tid':  ajaxify.data.tid,
+					'uid': app.user.uid,
```

Comment:
No need to pass the uid to the server. Server side already has the current users uid in socket.uid.